MaxText/tokenizer.py

```diff
@@ -17,35 +17,55 @@
 """Provides op for tokenizing a dataset."""
 
 from typing import Any, Dict, Iterable
-
 import dataclasses
-from absl import logging
 import tensorflow as tf
 import tensorflow_text as tftxt
-
+from jetstream.third_party.llama3 import llama3_tokenizer
 import max_logging
 
 
 Features = Dict[str, tf.Tensor]
 
 
-def _load_sentencepiece_tokenizer(tokenizer_path: str, add_bos: bool = False, add_eos: bool = True, reverse: bool = False):
-  """Load a tf-text SentencePiece tokenizer from given model filepath."""
-  max_logging.log(f"Tokenizer path: {tokenizer_path}")
-  with tf.io.gfile.GFile(tokenizer_path, "rb") as model_fp:
-    sp_model = model_fp.read()
-  sp_tokenizer = tftxt.SentencepieceTokenizer(model=sp_model, add_bos=add_bos, add_eos=add_eos, reverse=reverse)
-  return sp_tokenizer
+class TikToken():
```

Comment:
qq; why not import and use this: https://github.com/google/JetStream/blob/main/jetstream/engine/token_utils.py#L353

Reply:
The reason I created a class for it and not used the jetstream is because of the error below. Our input pipeline is tfds based so we are tokenizing Symbolic Tensors instead of np array's 
```
File "/__w/maxtext/maxtext/MaxText/tokenizer.py", line 79, in __call__  *
        features[k], _ = self.sp_tokenizer.encode(str(features[k]), is_bos = self.add_bos, is_eos = self.add_eos)
    File "/usr/local/lib/python3.10/dist-packages/jetstream/engine/token_utils.py", line 271, in encode  *
        tokens = np.array(self.vocab.encode_tf(s))

    NotImplementedError: Cannot convert a symbolic tf.Tensor (SentenceTokenizer/SentenceTokenizer/SentencepieceTokenizeOp:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported
```

Do you suggest another way to get around this problem ?

Reply:
I don't think the Tiktoken class you have created can handle tf.Tensor as well. 

Looks like the SentencePieceTokenizer you used from tensorflow_text made it into a tf op. And I don't think tensorflow_text has tiktoken.

Modifying maxtext to pass in numpy arrays (for both tiktoken and sentencepiece) should be the way to go.